src/overlays/actors/ovl_Kago/ac_kago.c

```diff
@@ -17,25 +27,117 @@ ActorProfile Kago_Profile = {
     /* */ GAMEPLAY_KEEP,
     /* */ sizeof(Kago),
     /* */ aKAG_actor_ct,
-    /* */ func_80A8F340_jp,
+    /* */ aKAG_actor_dt,
     /* */ aKAG_actor_init,
     /* */ aKAG_actor_draw,
     /* */ NULL,
 };
-#endif
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Kago/ac_kago/aKAG_actor_ct.s")
+void aKAG_actor_ct(Actor* thisx, UNUSED Game_Play* game_play) {
+    Kago* this = THIS;
+
+    this->type = this->actor.fgName - 0x5835;
+    this->objectType = this->type + 0x23;
+    this->paletteType = this->type + 0x4C;
```

Comment:
"objectType" is an enum value for a table of all the different structures. so I feel like structureType and structurePalette are better names. "type" is also apparently "action" in the gamecube repo, but eventually all the structure actors will use a shared struct so its fine not to name it right now.